schema_based_on_diagram.sql

```diff
@@ -0,0 +1,91 @@
+CREATE TABLE IF NOT EXISTS public.patients
+(
+    id integer,
+    name character varying,
+    date_of_birth date,
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE IF NOT EXISTS public.medical_histories
+(
+    id integer,
+    admiited_at timestamp with time zone,
+    patient_id integer,
+    status character varying,
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE IF NOT EXISTS public.invoices
+(
+    id integer,
+    total_amount numeric,
+    generated_at timestamp with time zone,
+    payed_at time with time zone,
+    medical_history_id integer,
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE IF NOT EXISTS public.invoice_items
+(
+    id integer,
+    unit_price numeric,
+    quantity integer,
+    total_price numeric,
+    invoice_id integer,
+    treatment_id integer,
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE IF NOT EXISTS public.treatments
+(
+    id integer,
+    type character varying,
+    name character varying,
+    PRIMARY KEY (id)
+);
+
+ALTER TABLE IF EXISTS public.medical_histories
+    ADD CONSTRAINT patient_id FOREIGN KEY (patient_id)
+    REFERENCES public.patients (id) MATCH SIMPLE
+    ON UPDATE NO ACTION
+    ON DELETE NO ACTION
+    NOT VALID;
+ALTER TABLE IF EXISTS public.medical_histories
+    ADD FOREIGN KEY (id)
+    REFERENCES public.treatments (id) MATCH SIMPLE
+    ON UPDATE NO ACTION
+    ON DELETE NO ACTION
+    NOT VALID;
```

Comment:
- [ ] Please create a new table `medical_histories_treatments` to address the many-to-many relationship between `medical_histories` and `treatments` tables and add the foreign keys in it:
```sql
 CREATE TABLE medical_histories_treatments (
    medical_history_id int references medical_histories(id),
    treatment_id int references treatments(id),
  );
```